datafusion/src/logical_plan/expr.rs

```diff
@@ -1598,7 +1611,7 @@ impl fmt::Debug for Expr {
             Expr::IsNull(expr) => write!(f, "{:?} IS NULL", expr),
             Expr::IsNotNull(expr) => write!(f, "{:?} IS NOT NULL", expr),
             Expr::BinaryExpr { left, op, right } => {
-                write!(f, "{:?} {:?} {:?}", left, op, right)
+                write!(f, "{:?} {} {:?}", left, op, right)
```

Comment:
@Dandandan FYI i updated here to use display and i think thats what made it work.  i was able to get the expected results on some tests after this as well.

Are you ok with this approach?

Reply:
I would be ok with this as intermediate solution. Ideally we'll move the formatting to `Display` instead and use that for printing @alamb wdyt?

Reply:
I agree with @Dandandan